tapdb/addrs.go

```diff
@@ -467,7 +467,9 @@ func (t *TapAddressBook) QueryAddrs(ctx context.Context,
 				return fmt.Errorf("unable to make addr: %w", err)
 			}
 
-			declaredKnown := addr.ScriptKeyDeclaredKnown.Valid
+			declaredKnown := extractBool(
```

Comment:
Oops, don't know what I was thinking at that moment... Nice catch!

Reply:
Thx! Likely need to do some more analysis, but in most cases this would've just resulted in an RPC display being wrong. There is the case though when we decide to create a transfer output or not, that relied on this erroneous check.